plugins/kitpvp-paper/src/main/java/dev/drekamor/warp/database/DatabaseManager.java

```diff
@@ -0,0 +1,135 @@
+                name VARCHAR(64) UNIQUE PRIMARY KEY NOT NULL,
+                world VARCHAR(64) NOT NULL,
+                x REAL NOT NULL,
+                y REAL NOT NULL,
+                z REAL NOT NULL,
+                pitch REAL NOT NULL,
+                yaw REAL NOT NULL,
+                gamemode VARCHAR(16) NOT NULL
+            );
+            """);
+
+    }
+
+    public Map<String, Warp> getWarps() {
+
+        Map<String, Warp> warps = new HashMap<>();
+        try (Connection connection = this.getConnection();){
+            PreparedStatement statement = connection.prepareStatement("SELECT * FROM warps;");
+            ResultSet results = statement.executeQuery();
+            while(results.next()) {
+                warps.put(
+                        results.getString("name"),
+                        new Warp(
+                                results.getString("name"),
+                                results.getString("world"),
+                                results.getDouble("x"),
+                                results.getDouble("y"),
+                                results.getDouble("z"),
+                                results.getFloat("pitch"),
+                                results.getFloat("yaw"),
+                                results.getString("gamemode")
+                        )
+                );
+            }
+            connection.close();
+        } catch (SQLException e) {
+            plugin.getLogger().severe("Failed to retrieve warps");
+            plugin.getLogger().severe(Arrays.toString(e.getStackTrace()));
+        }
+        return warps;
+    }
+
+    public List<String> getWarpNames() {
+        List<String> warps = new ArrayList<>();
+        try(Connection connection = this.getConnection()) {
+            ResultSet results = connection.prepareStatement("SELECT name FROM warps;").executeQuery();
+            while (results.next()) {
+                warps.add(results.getString("name"));
+            }
+            connection.close();
+        } catch (SQLException e) {
+            plugin.getLogger().severe("Failed to retrieve warp names");
+            plugin.getLogger().severe(Arrays.toString(e.getStackTrace()));
+        }
+        return warps;
+    }
+
+    public boolean addWarp(Warp warp) {
+        try (Connection connection = this.getConnection()){
+            PreparedStatement statement = connection.prepareStatement("INSERT IGNORE INTO warps(name, world, x, y, z, pitch, yaw, gamemode) VALUES (?, ?, ?, ?, ?, ?, ?, ?)");
+            statement.setString(1, warp.name());
+            statement.setString(2, warp.world());
+            statement.setDouble(3, warp.x());
+            statement.setDouble(4, warp.y());
+            statement.setDouble(5, warp.z());
+            statement.setFloat(6, warp.pitch());
+            statement.setFloat(7, warp.yaw());
+            statement.setString(8, warp.gamemode());
+            statement.execute();
```

Comment:
You can use executeUpdate here and checked if the number of updated rows was 0 or 1 to see if the insert was actually successful. Since you are using INSERT IGNORE you could be doing nothing (that is, trying to add a warp with the same name as an existing warp) and this method will still return true